recipes/sol2/3.x.x/conanfile.py

```diff
@@ -0,0 +1,69 @@
+from conans import ConanFile, tools, CMake
+from conans.errors import ConanInvalidConfiguration, ConanException
+import os
+
+
+class Sol2Conan(ConanFile):
+    name = "sol2"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/ThePhD/sol2"
+    description = "C++17 Lua bindings"
+    topics = ("conan", "lua", "c++", "bindings")
+    settings = "os", "compiler"
+    license = "MIT"
+    requires = ["lua/5.3.5"]    
+    default_options = {"lua:compile_as_cpp": True}
+    
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+    _cmake = None
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+    
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.configure(
+            source_folder=self._source_subfolder,
+            build_folder=self._build_subfolder
+        )
+        return self._cmake
+
+
+    def configure(self):
+        minimal_cpp_standard = "17"
+        try:
+            tools.check_min_cppstd(self, minimal_cpp_standard)
+        except ConanInvalidConfiguration:
+            raise
+        except ConanException:
+            # FIXME: We need to handle the case when Conan doesn't know
+            # about a user defined compiler's default standard version
+            self.output.warn(
+                "Unnable to determine the default standard version of the compiler")
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+    # this is buggy in < 3.2.0, and it is less work to just copy the headers, 
+    # because this is what install does. However, for future releases, leave this here    
+#        cmake = self._configure_cmake()
+#        cmake.install()
+#        tools.rmdir(os.path.join(self.package_folder, "share")) # constains just # , "pkgconfig"))
+#        tools.rmdir(os.path.join(self.package_folder, "lib" )) # constains just # , "cmake"))
+
+        self.copy("*.h", src=os.path.join(self._source_subfolder, "include"), dst="include", keep_path=True)
+        self.copy("*.hpp", src=os.path.join(self._source_subfolder, "include"), dst="include", keep_path=True)
+        self.copy("LICENSE.txt", src=self._source_subfolder, dst="licenses")
+
+    def package_id(self):
+        self.info.header_only()
+
+    def package_info(self):        
+        self.cpp_info.defines.append("SOL_USING_CXX_LUA=1")
```

Comment:
I see you've put this here.

Can't you just do this?

```suggestion
        self.cpp_info.defines.append("SOL_USING_CXX_LUA={}".format(1 if self.options["lua"].compile_as_cpp else 0)
```

And remove the `default_option` at the top of this file.

Reply:
Ooh. Yes please, make it an option!

Reply:
No options should be added. The option of `lua` is re-used. (If this works)

Reply:
Oh, yes. Sorry, I meant, use the lua option, propagate it into sol2, please. :D

Reply:
yes, I can query the option from lua, of cause
in this case we do not need any define at all, I can add that
but it will take some time since I test it an 14 configurations * 2, and this on a macbook , not fun ;-) 
It seems I should start activate my own server side builds with github actions or azure to accelerate that ...